pkg/command/map_string.go

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of Cilium
+
+package command
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cast"
+	"github.com/spf13/viper"
+)
+
+var keyValueRegex = regexp.MustCompile(`((\w+)=(\w+,))*((\w+)=(\w+))$`)
+
+// GetStringMapString contains one enhancement to support k1=v2,k2=v2 compared to original
+// implementation of GetStringMapString function
+// Related upstream issue https://github.com/spf13/viper/issues/911
+func GetStringMapString(vp *viper.Viper, key string) map[string]string {
+	v, _ := GetStringMapStringE(vp, key)
+	return v
+}
+
+// GetStringMapStringE is same as GetStringMapString, but with error
+func GetStringMapStringE(vp *viper.Viper, key string) (map[string]string, error) {
```

Comment:
Why are we returning an initialized `map[string]string{}` for the error cases?

Reply:
this is to be aligned with upstream function.

https://github.com/spf13/cast/blob/master/caste.go#L863